File: app/src/main/java/com/rohitsuratekar/NCBSinfo/common/AppPrefs.java
package com.rohitsuratekar.NCBSinfo.common;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class AppPrefs {

    private static final String FAVORITE_ORIGIN = "favorite_origin";
    private static final String FAVORITE_DESTINATION = "favorite_destination";
    private static final String FAVORITE_TYPE = "favorite_type";
    private static final String EGG_ACTIVE = "egg_active";
    private static final String NOTIFICATIONS = "notifications";
    private static final String SETTINGS_DEFAULT_SET = "settings_default_set";
    private static final String LOCATION_SORT = "location_sort";
    private static final String DEVELOPER_ACTIVE = "developer_active";
    private static final String ADMIN_CODE = "admin_code";
    private static fi


    private SharedPreferences prefs;

    public AppPrefs(Context context) {
        this.prefs = PreferenceManager.getDefaultSharedPreferences(context);
    }


    public String getFavoriteOrigin() {
        return get(FAVORITE_ORIGIN, "ncbs");
    }

    public String getFavoriteDestination() {
        return get(FAVORITE_DESTINATION, "iisc");
    }

    public String getFavoriteType() {
        return get(FAVORITE_TYPE, "shuttle");
    }

    public void setFavoriteOrigin(String origin) {
        put(FAVORITE_ORIGIN, origin);
    }

    public void setFavoriteDestination(String destination) {
        put(FAVORITE_DESTINATION, destination);
    }

    public void setFavoriteType(String type) {
        put(FAVORITE_TYPE, type);
    }

    public boolean isEggActive() {
        return get(EGG_ACTIVE, false);
    }

    public void eggActivated() {
        put(EGG_ACTIVE, true);
    }

    public void removeEggs() {
        put(EGG_ACTIVE, false);
    }

    public boolean areNotificationsAllowed() {
        return get(NOTIFICATIONS, true);
    }

    public void toggleNotifications() {
        put(NOTIFICATIONS, !get(NOTIFICATIONS, true));
    }

    public void setNotificationAllowed(boolean value) {
        put(NOTIFICATIONS, value);
    }

    public boolean isDefaultRouteSet() {
        return get(SETTINGS_DEFAULT_SET, false);
    }

    public void defaultRouteSet() {
        put(SETTINGS_DEFAULT_SET, true);
    }

    public int getLocationSort() {
        return get(LOCATION_SORT, 0);
    }

    public void setLocationSort(int loc) {
        put(LOCATION_SORT, loc);
    }

    public void setDeveloperActive() {
        put(DEVELOPER_ACTIVE, true);
    }

    public void removeDeveloper() {
        put(DEVELOPER_ACTIVE, false);
    }

    public boolean isDeveloperActive() {
        return get(DEVELOPER_ACTIVE, false);
    }

    public String getAdminCode() {
        return get(ADMIN_CODE, "");
    }

    public void setAdminCode(String code) {
        put(ADMIN_CODE, code);
    }

    private String get(String key, String defaultValue) {
        return prefs.getString(key, defaultValue);
    }

    private int get(String key, int defaultValue) {
        return prefs.getInt(key, defaultValue);
    }

    private boolean get(String key, boolean defaultValue) {
        return prefs.getBoolean(key, defaultValue);
    }

    private void put(String key, String value) {
        prefs.edit().putString(key, value).apply();
    }

    private void put(String key, boolean value) {
        prefs.edit().putBoolean(key, value).apply();
    }

    private void put(String key, int value) {
        prefs.edit().putInt(key, value).apply();
    }


}

File: app/src/main/java/com/rohitsuratekar/NCBSinfo/common/AppPrefs.kt
package com.rohitsuratekar.NCBSinfo.common

import android.content.Context
import android.preference.PreferenceManager

private const val FAVORITE_ORIGIN = "favorite_origin"
private const val FAVORITE_DESTINATION = "favorite_destination"
private const val FAVORITE_TYPE = "favorite_type"
private const val NOTIFICATIONS = "notifications"
private const val LOCATION_SORT = "location_sort"
private const val UPDATE_VERSION = "update_version"
private const val CRASH_REPORTING = "crash_reporting"
private const val CRASH_REPORTING_REQUEST = "crash_reporting_request"

class AppPrefs(context: Context) {
    private val pref = PreferenceManager.getDefaultSharedPreferences(context)

    fun clearAll() {
        pref.edit().clear().apply()
    }

    fun favoriteOrigin(): String {
        return pref.getString(FAVORITE_ORIGIN, "ncbs") ?: kotlin.run { return "ncbs" }
    }

    fun favoriteOrigin(origin: String) {
        pref.edit().putString(FAVORITE_ORIGIN, origin).apply()
    }

    fun favoriteDestination(): String {
        return pref.getString(FAVORITE_DESTINATION, "iisc") ?: kotlin.run { return "iisc" }
    }

    fun favoriteDestination(destination: String) {
        pref.edit().putString(FAVORITE_DESTINATION, destination).apply()
    }

    fun favoriteType(): String {
        return pref.getString(FAVORITE_TYPE, "shuttle") ?: kotlin.run { return "shuttle" }
    }

    fun favoriteType(type: String) {
        pref.edit().putString(FAVORITE_TYPE, type).apply()
    }

    fun isNotificationAllowed(): Boolean {
        return pref.getBoolean(NOTIFICATIONS, true)
    }

    fun locationSort(): Int {
        return pref.getInt(LOCATION_SORT, 0)
    }

    fun locationSort(location: Int) {
        pref.edit().putInt(LOCATION_SORT, location).apply()
    }

    fun updateVersion(): Int {
        return pref.getInt(UPDATE_VERSION, -1)
    }

    fun updateVersion(version: Int) {
        pref.edit().putInt(UPDATE_VERSION, version).apply()
    }

    fun crashReportingEnabled(): Boolean {
        return pref.getBoolean(CRASH_REPORTING, false)
    }

    fun crashReportingEnabled(value: Boolean) {
        pref.edit().putBoolean(CRASH_REPORTING, value).apply()
    }

    fun showCrashReportingRequest(): Boolean {
        return pref.getBoolean(CRASH_REPORTING_REQUEST, true)
    }

    fun showCrashReportingRequest(value: Boolean) {
        pref.edit().putBoolean(CRASH_REPORTING_REQUEST, value).apply()
    }
}